core/coreapi/block.go

```diff
@@ -38,13 +38,29 @@ func (api *BlockAPI) Put(ctx context.Context, src io.Reader, opts ...caopts.Bloc
 	var pref cid.Prefix
 	pref.Version = 1
 
+	if settings.Codec == "" {
+		if settings.MhType != mh.SHA2_256 || (settings.MhLength != -1 && settings.MhLength != 32) {
+			settings.Codec = "protobuf"
+		} else {
+			settings.Codec = "v0"
+		}
+	}
+
```

Comment:
Should this be in `BlockPutOptions` instead, it seems like it is options handling.

Reply:
This way we separate Options validation from implementation.